website_seo_redirection/models/website_seo_redirection.py

```diff
@@ -42,6 +54,21 @@ def _check_destination(self):
         self._url_format_check("destination")
 
     @api.multi
+    @api.constrains("origin", "destination")
+    def _check_not_recursive(self):
+        """Avoid infinite loops."""
+        for redirection in self:
+            origins = {redirection.origin}
+            while redirection:
+                redirection = self.search([
+                    ("origin", "=", redirection.destination),
+                ])
+                if redirection.origin in origins:
+                    raise ValidationError(
+                        _("Recursive redirection is forbidden."))
+                origins.add(redirection.origin)
```

Comment:
Won't this fail on the last iteration? Looking at the while loop, it will terminate when not `redirection` which would mean that at some point we're trying a `.origin` on a `False` or `None`?